py-rattler/rattler/solver/solver.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+from typing import List, Tuple
+from rattler.match_spec.match_spec import MatchSpec
+
+from rattler.rattler import py_solve
+from rattler.repo_data.record import RepoDataRecord
+from rattler.virtual_package.generic import GenericVirtualPackage
+from rattler.repo_data.repo_data import RepoData
+from rattler.channel import Channel
+
+
+def solve(
+    specs: List[MatchSpec],
+    available_packages: List[Tuple[RepoData, Channel]],
+    locked_packages: List[RepoDataRecord],
+    pinned_packages: List[RepoDataRecord],
+    virtual_packages: List[GenericVirtualPackage],
+) -> List[RepoDataRecord]:
```

Comment:
It would be nice to use default values for the named arguments so people don't have to fill it in if they don't know what the arguments mean. E.g. `locked_packages: Optional[List[RepoDataRecord]] = None`